src/static/helpers/slasHelper.ts

```diff
@@ -141,22 +152,25 @@ export async function authorize(
     redirect: isBrowser ? 'follow' : 'manual',
   };
 
-  const options = {
+  const {hint, redirectURI, usid, ...restOfParams} = parameters;
+  const opts = {
     parameters: {
       client_id: slasClient.clientConfig.parameters.clientId,
       channel_id: slasClient.clientConfig.parameters.siteId,
       ...(clientOptions.codeChallenge && {
         code_challenge: clientOptions.codeChallenge,
       }),
-      ...(parameters.hint && {hint: parameters.hint}),
+      ...(hint && {hint}),
       organizationId: slasClient.clientConfig.parameters.organizationId,
-      redirect_uri: parameters.redirectURI,
+      redirect_uri: redirectURI,
       response_type: 'code',
-      ...(parameters.usid && {usid: parameters.usid}),
+      ...(usid && {usid}),
+      ...restOfParams,
     },
+    ...(options?.headers || {}),
```

Comment:
headers have to be defined under the `headers` property
```suggestion
    ...(options?.headers && { headers: options.headers }),
```

Reply:
If the tests didn't catch this, then we have a problem with the tests...

Reply:
This is my mistake. They will treat these params as custom params for parameters objects, and it is supposed to throw a warning here if there are an invalid params. I added a test for that. Thanks for pointing this out!